### ISSUE TITLE
Backport borg (borgbackup) 1.1+ to 17.09?

### ISSUE THREAD

**Issue by primeos** [NixOS/nixpkgs#33053]
## Issue description

It seems like there where security relevant changes ([Fix AES-CTR nonce reusage](https://github.com/borgbackup/borg/pull/1374)) that we should backport to the stable branch but apparently upstream failed to mention this in the release notes.

When I create a backup from a client with version 1.1.1 to a remote server with version 1.0.11 I get the following warning:
```
Remote: Borg 1.0.11: exception in RPC call:                                                                                                                                                                        
Remote: Traceback (most recent call last):
Remote:   File "/nix/store/2k0prf34z4jbj4pv3mm7j4gd9zd2chyq-borgbackup-1.0.11/lib/python3.6/site-packages/borg/remote.py", line 141, in serve
Remote:     raise InvalidRPCMethod(method)
Remote: borg.remote.InvalidRPCMethod: RPC method get_free_nonce is not valid
Remote: Platform: Linux xyz 4.9.27 #1-NixOS SMP Mon May 8 05:48:32 UTC 2017 x86_64
Remote: Linux:   
Remote: Borg: 1.0.11  Python: CPython 3.6.4
Remote: PID: 11136  CWD: /home/michael
Remote: sys.argv: ['/nix/store/2k0prf34z4jbj4pv3mm7j4gd9zd2chyq-borgbackup-1.0.11/bin/borg', 'serve', '--umask=077']
Remote: SSH_ORIGINAL_COMMAND: None
Remote: 
Please upgrade to borg version 1.1+ on the server for safer AES-CTR nonce handling.
```

According to the [changelog](https://github.com/borgbackup/borg/blob/1.1.3/docs/changes.rst#version-113-2017-11-27) we should probably backport version 1.1.3 as it contains security fixes as well or manually apply the patch to a 1.1+ release.

cc @NixOS/security-notifications @nckx @flokli

**Comment by flokli** [comment]
As discussed with @ThomasWaldmann on irc, we should backport borgbackup 1.1.4 (from https://github.com/NixOS/nixpkgs/pull/33309)

**Comment by primeos** [comment]
Fixed by #34368 which is now in `nixpkgs-channels/nixos-17.09`.
